### DISCUSSION TITLE
Idea - Recovery Pallet

### DISCUSSION THREAD

**Discussion by Arn0-js** [capsule-corp-ternoa/ternoa-js#95] [Ideas]
Hi team,

Being curious but also anxious about my ability to preserve seed phrases over the next years or decades, I found this interesting [pallet_recovery](https://paritytech.github.io/substrate/master/pallet_recovery/index.html#).

Its name speaks for itself, but for those who don't know, here is its summary : 
The Recovery pallet is an M-of-N social recovery tool for users to gain access to their accounts if the private key or other authentication mechanism is lost. Through this pallet, a user is able to make calls on-behalf-of another account which they have recovered. The recovery process is protected by trusted “friends” whom the original account owner chooses. A threshold (M) out of N friends are needed to give another account access to the recoverable account.

There are naturally pros and cons (risks), but the procedure is smartly implementend (countdown and consent aspect, similarities with the Secret Capsule concepts actually... ). 

Next to the capsule product/app, this could bring added value to the Ternoa Wallet. 
What's your point of view ? 

Thanks,

Arn0

**Comment by markopoloparadox** [comment]
Hello,

This is seems like a good idea. Usually it doesn't take too much time to implement and configure already available standard substrate pallets into our runtime. It might take some time for the SDK and wallet to implement this but we will see.

@peshwar9 @ipapandinas @dhammani @M2613  what do you think about this?

**Reply by jeemclr** [reply]
Interesting and sounds like "Capsule NFT" with the consent protocol (trust friends). Maybe looking into that direction.

**Comment by Leouarz** [comment]
Great idea !

**Comment by ipapandinas** [comment]
In fact this is a great idea!
I can see Wallet being the perfect dApp for this feature.
The full implementation of this pallet in the SDK is not the priority for the moment. This is not a blocking factor as you can could use generic helpers like `createTx`, `signTx` and `submitTx` to interact with.
